test/libsolidity/semanticTests/abiEncoderV1/memory_dynamic_array_and_calldata_bytes.sol

```diff
@@ -0,0 +1,17 @@
+pragma abicoder v1;
+
+contract C {
+    function f(uint256[] memory a, bytes calldata b) public returns (bytes memory) {
+        return abi.encode(a, b);
+    }
+
+    function g(uint256[] memory a, bytes calldata b) external returns (bytes memory) {
+        return f(a, b);
+    }
+}
```

Comment:
> I added a test case for v1, the problem couldn't be reproduced. Maybe that's why it didn't show up for so long?

If you meant this case, then I I think it does not reproduce it simply because the static array is not the last argument.

This case actually makes it overwrite stuff:
```solidity
pragma abicoder v1;

contract C {
    function f(bool a, bytes calldata b, bytes32[2] calldata c)
        public
        returns (bool, bytes calldata, bytes32[2] calldata)
    {
        return (a, b, c);
    }
}
// ----
// f(bool,bytes,bytes32[2]): true, 0x80, "a", "b", 4, "abcd" -> true, 0x80, "a", "b", 4, "abcd"
```
```
  [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1]
  [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,80]
  [61,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
  [62,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
  [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
  [61,62,63,64,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
```

@ekpyron Since these are dynamic types, is it actually going through v1 or is it just that v2 gets force-enabled in some way?

Reply:
abiencoder v1 does have dynamic types, just not nested ones, so this should work with v1.
From a quick look at it, running this through ``isoltest`` will still try to run it via-IR (which implies abiencoder v2) and that's what's failing - adding ``compileViaYul: false`` (i.e. having it really *only* run with abiencoder v1) has it pass.

Reply:
Right. Thanks.

In that case we should add this as a use case for v1 (with the extra `compileViaYul: false`) but fortunately v1 looks unaffected.